server/pulp/plugins/util/publish_step.py

```diff
@@ -1083,40 +1106,14 @@ def process_main(self):
         # any units that are already in pulp
         units_we_already_had = set()
 
-        # mongodb throws exceptions for too big queries, so we spilt it into multiple smaller ones
-        for page in misc.paginate(self.parent.available_units, 50):
-            # for any unit that is already in pulp, save it into the repo
-            for unit_dict in self.content_query_manager.get_multiple_units_by_keys_dicts(
-                    self.unit_type, page, self.unit_key_fields):
-                unit = self._dict_to_unit(unit_dict)
-                self.get_conduit().save_unit(unit)
-                units_we_already_had.add(unit)
+        for units_group in misc.paginate(self.parent.available_units, 50):
```

Comment:
Consider moving the `50` to a constant. I know you didn't introduce that with this change. Not a requirement.